src/screens/productDetails/ProductDetailsScreen.tsx

```diff
@@ -0,0 +1,182 @@
+import React, {FC, useCallback, useRef, useState} from 'react'
+import {
+  RefreshControl,
+  SafeAreaView,
+  ScrollView,
+  Dimensions,
+  View,
+  Image,
+  StyleSheet,
+  Text,
+} from 'react-native'
+import Carousel, {Pagination} from 'react-native-snap-carousel'
+import Icon from 'react-native-vector-icons/FontAwesome'
+import {Button, Divider} from 'react-native-elements'
+import {ProductModel, useGetProductByIdQuery} from '../../features/products'
+import {NavigationInterface} from '../../shared/types/navigation.interface'
+import {productImagesMocks as images} from './images.mocks'
+
+type RenderCarouselItem = {
+  item: any
+  index: number
+}
+
+const renderCarouselItem = ({item}: RenderCarouselItem) => {
+  const {src} = item
+  return <Image source={src} />
+}
+
+interface ProductDetailsScreenProps extends NavigationInterface {}
+
+export const ProductDetailsScreen: FC<ProductDetailsScreenProps> = ({
+  route,
+}) => {
+  const {data, isLoading, refetch} = useGetProductByIdQuery(route.params.id)
+  const product: ProductModel | undefined = data?.data
+  const [currentCarouselIndex, setCurrentCarouselIndex] = useState<number>(0)
+  const {width: viewportWidth} = Dimensions.get('window')
+  const carouselRef = useRef<HTMLElement | any>()
+
+  const onRefresh = useCallback(() => refetch(), [])
+  const refHandler = (ref: HTMLElement) => {
+    carouselRef.current = ref
+  }
+  const onSnapHandler = (index: number) => setCurrentCarouselIndex(index)
+  const onNextHandler = (): void => {
```

Comment:
how you decide whatever to useCallback or not?

Reply:
It's more difficult to decide when you use a component from a third-party library, in this case I would add useCallback for ```onSnapHandler, onNextHandler , onPreviousHandler ``` handlers as well since we pass the handlers to custom components which might re-render under certain circumstances.

What do you think it is worth to use useCallback for ```onSnapHandler, onNextHandler , onPreviousHandler``` handlers?